compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/MockDesugar.java

```diff
@@ -115,35 +115,13 @@ public void generateMockFunctions(BLangPackage pkgNode) {
         // Get the Mock Function map from the pkgNode
         Map<String, String> mockFunctionMap = pkgNode.getTestablePkg().getMockFunctionNamesMap();
 
-        // Get all the imports symbols from the testable package
-        Set<String> mockFunctionSet = mockFunctionMap.keySet();
-        ArrayList<String> importsList = new ArrayList<>();
-        for (BLangImportPackage importPkg : pkgNode.getTestablePkg().getImports()) {
-            if (importPkg.symbol == null) {
-                continue;
-            }
-            if (!importPkg.symbol.toString().contains(testPackageSymbol)) {
-                importsList.add(importPkg.symbol.toString());
-            }
-        }
-
-        // Get all the imports from the current package
-        for (BLangImportPackage importPkg : pkgNode.getImports()) {
-            if (importPkg.symbol == null) {
-                continue;
-            }
-            if (importsList.contains(importPkg.symbol.toString())) {
-                continue;
-            }
-            if (!importPkg.symbol.toString().contains(testPackageSymbol)) {
-                importsList.add(importPkg.symbol.toString());
-            }
-        }
+        // Get the mock function type map from the pkgNode
+        Map<String, Boolean> isLegacyMockingMap = pkgNode.getTestablePkg().getIsLegacyMockingMap();
```

Comment:
I feel like the name `isLegacyMockingMap` feels as if it is a boolean. Instead if we make an enum, and make the variable `mockTypes` as below, it would be cleaner.
``` java
enum MockingType {
   LEGACY, NEW
}

private Map<String, MockType> mockTypes;
```

Reply:
we can use `enum`. But, the map above is accessed by multiple classes. We need to define this `enum` as `public`  and need to move to a separate file. As the use-case of this map is limited. I feel, that is quite unnecessary.